### DISCUSSION TITLE
Is there any python library for calling C API directly from python?

### DISCUSSION THREAD

**Discussion by wolvever** [triton-inference-server/server#5897] [Q&A]
To call Triton ensemble models without grpc call, one way is using C API.  Is there any Python library so that I can reuse existing ensemble model config files? 

I looked at PyTriton and its documentation is focusing on binding python functions into Triton, while my problem is not related to bind python functions into Triton easily.  My models are Paddle models and C backend code.  That make Triton server a good investment which can be use to serve ensemble models.  

Now I'm trying to use Triton combined with RayServe,  using python triton GRPC client will incur more grpc call overhead, and not compatible with RayServe auto scaling.  So the best way is to use C API in Python ray actor that requires a Python library for calling Triton C API.

**Comment by duyleekun** [comment]
Check this out

https://github.com/triton-inference-server/tutorials/tree/main/Triton_Inference_Server_Python_API
